pkg/controllers/vdb/status_reconciler.go

```diff
@@ -182,6 +183,24 @@ func (s *StatusReconciler) calculateClusterStatus(stat *vapi.VerticaDBStatus) {
 	}
 }
 
+// getSubclusterStatusType returns the subcluster status type depends on its type in subclusters and sandboxes
+func (s *StatusReconciler) getSubclusterStatusType(podName types.NamespacedName) string {
```

Comment:
```
func (s *StatusReconciler) getSubclusterStatusType(podName types.NamespacedName) string {
	if !s.PFacts.DoesDBExist() {
		return ""
	}

	sandboxStatus := s.Vdb.GetSandboxStatus(s.PFacts.SandboxName)
	isSandbox := sandboxStatus != nil && sandboxStatus.Name != vapi.MainCluster

	if s.PFacts.Detail[podName].GetIsPrimary() {
		if isSandbox {
			return vapi.SandboxPrimarySubcluster
		}
		return vapi.PrimarySubcluster
	}

	if isSandbox {
		return vapi.SandboxSecondarySubcluster
	}

	return vapi.SecondarySubcluster
}

```

Reply:
Thanks! Also updated to pass sc instead of the podName as the parameter.